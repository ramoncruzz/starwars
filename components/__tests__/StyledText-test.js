import * as React from 'react';
import renderer from 'react-test-renderer';
import { FlatList } from 'react-native';

import { MonoText } from '../StyledText';
import ScreenStarWars from '../../src/screens/ScreenB/TabTwoScreen';
import { Page1 } from '../../src/hooks/mocks/people.pages';

let mockPeopleList = Page1.results;
let mockNextPage = jest.fn();
let mockRemovePerson = jest.fn();

jest.mock('../../src/hooks/useStarWars', () => {
  return jest.fn().mockReturnValue({
    peopleList: mockPeopleList,
    nextPage: mockNextPage,
    removePerson: mockRemovePerson,
  });
});

it(`renders correctly`, () => {
  const tree = renderer.create(<MonoText>Snapshot test!</MonoText>);

  expect(tree).toMatchSnapshot();
});

it(`renders Star Wars screen`, () => {
  const tree = renderer.create(<ScreenStarWars />).toJSON();
  expect(tree).toMatchSnapshot();
});
