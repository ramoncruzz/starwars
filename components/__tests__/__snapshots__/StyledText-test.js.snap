// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Star Wars screen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;

exports[`renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
      },
      Array [
        undefined,
        Object {
          "fontFamily": "space-mono",
        },
      ],
    ]
  }
>
  Snapshot test!
</Text>
`;
